internal/ingest/httpingest/httpingest.go

```diff
@@ -26,42 +28,84 @@ type Collector interface {
 
 func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	logger := h.getLogger()
+	h.Context = r.Context()
+
+	contentType := r.Header.Get("Content-Type")
+
+	var err error
+	switch contentType {
+	case "application/cloudevents+json":
+		err = h.processSingleRequest(w, r)
+	case "application/cloudevents-batch+json":
+		err = h.processBatchRequest(w, r)
+	default:
+		_ = render.Render(w, r, api.ErrUnsupportedMediaType(errors.New("content type must be application/cloudevents+json or application/cloudevents-batch+json")))
+	}
+
+	if err != nil {
+		logger.ErrorCtx(h.Context, "unable to process request", "error", err)
+		_ = render.Render(w, r, api.ErrInternalServerError(err))
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h Handler) processBatchRequest(w http.ResponseWriter, r *http.Request) error {
+	var events []event.Event
+
+	err := json.NewDecoder(r.Body).Decode(&events)
+	if err != nil {
+		return err
+	}
+
+	for _, event := range events {
+		err = h.processEvent(event)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This is probably OK for now, but generally batch ingestion endpoints are supposed to be able to process events asynchronously.